### ISSUE TITLE
Meeting: October 2, 2019

### ISSUE THREAD

**Issue by Lokathor** [rust-gamedev/wg#55]
## Action Items From Last Meeting

* [ ] Discuss open `wg` PRs
* [ ] Investigate more about Placement New (pre-rfc? rfc? where is the ecosystem at).
* [ ] More chat about the state of rust audio libraries? Bindings to opens OpenAL, `cpal` latency issues (low-handing fruit?), the `cpal`/`rodio` workload split.
* [ ] Rust on Mobile (CI, cargo integration, etc)
* [ ] custom allocator followup ([wg link](https://github.com/rust-lang/wg-allocators))
* [ ] Any movement on the ecosystem guide (@Wodann)
* [ ] Improve issue tracker triage (smaller issues if possible, more labels about areas/skillsets, etc)
* [ ] Ecosystem survey (if it's been published).

**Comment by aclysma** [comment]
I won't be around tomorrow, but I feel like I should mention something regarding placement new since I opened the issue and it's on the agenda.

I'm not blocked by not having this feature, nor am I convinced this is a "must fix," even if it's a feature I'd like to have. (I'm a fan of static allocating object pools in order to avoid dynamic allocation.)

I left some thoughts in https://github.com/rust-gamedev/wg/issues/48

**Comment by kvark** [comment]
Status of raw-window-handle?

**Comment by Lokathor** [comment]
it says i'm not allowed to join the call >_>

**Comment by Wodann** [comment]
@kvark Having the same problem as @Lokathor

**Comment by Lokathor** [comment]
We'll use <https://hangouts.google.com/call/4qS8qRXEEQVQO-8ZWsV9AEEE> today I guess

**Comment by kvark** [comment]
If we have extra time, https://gfx-rs.github.io/2019/10/01/update.html

**Comment by Lokathor** [comment]
https://github.com/mtak-/rust-on-mobile

**Comment by kvark** [comment]
Move https://github.com/amethyst/amethyst/wiki/Other-Game-Engines-in-Rust to this repo

**Comment by Lokathor** [comment]
* gflw issue: https://github.com/PistonDevelopers/glfw-rs/issues/427
* winit 0.19 issue: https://github.com/rust-windowing/winit/issues/1202

**Comment by Wodann** [comment]
Meeting notes (10/02)

Attendance:
- @Lokathor 
- @kvark 
- @navi
- @mtak-
- @Wodann 
- Jake (Github handle?)

(If I am missing your Github handle and you attended, please react with a smiley)

Previous action items:
- Discuss open wg PRs:
  - @kvark 's ecosystem PR will be taken down once @Wodann opens a new one
  - it's unsure how to proceed with @aclysma 's known issues PR

- Investigate more about Placement New (pre-rfc? rfc? where is the ecosystem at).
  * @aclysma mentioned that this is not a "must fix" rather a "nice to have" (https://github.com/rust-gamedev/wg/issues/55#issuecomment-537336853)

- More chat about the state of rust audio libraries? Bindings to opens OpenAL, cpal latency issues (low-handing fruit?), the cpal/rodio workload split.
  * The maintainer of sepal mentioned that they are short-handed, but can give direction if someone were to help them solve the latency issue. (Discord chat)

- Rust on Mobile (CI, cargo integration, etc)
  * @mtak- has a demo setup for Rust on iOS/Android (https://github.com/mtak-/rust-on-mobile)
  * MaulingMonkey (Discord handle) might know more about Android setup. @Lokathor will reach out on Discord

- custom allocator followup (wg link)
  * If people need this functionality right now, people can create box-like types

- Any movement on the ecosystem guide (@Wodann)
  * @icefoxen posted a Reddit post about the state of Rust graphics programming that was strongly opinionated but it seemed very popular
  * for a more objective approach @Wodann will proceed with his prior proposal (https://github.com/rust-gamedev/rust-gamedev.github.io/issues/6). They'll come up with an example for gfx-hal and winit within the architecture overview and post it on the github.io website. From there on we'll discuss and decide on how to proceed

- Improve issue tracker triage (smaller issues if possible, more labels about areas/skillsets, etc)
  * Most of the work that happens doesn't happen within the wg repositories. As a result, updates might run out of sync.
  * It's unclear how to proceed

- Ecosystem survey (if it's been published).
  * Hasn't been assembled by @erlend-sh yet.

* gfx-rs update (@kvark): The new swapchain model has been simplified. This allows us to directly map it to Metal, OpenGL, etc. It does require some more work on Vulkan, but nothing serious. We recommend it for all users. (https://gfx-rs.github.io/2019/10/01/update.html)

* Move Amethyst's "Other Game Engines in Rust" article to the WG repo as addition to our ecosystem guide. (@kvark)
  - @Lokathor recommends tying this in with @Wodann 's proposal, or to at least take inspiration from it
  - it's super useful to find the projects that are using a particular crate as dependency but are not themselves published on crates.io (@kvark)

* raw-window-handle (@kvark)
  - It still requires glfw, sdl2, winit 0.19 support for it to be useful
  - sdl2 maintainers mentioned that they'd accept a PR that incorporates the raw-window-handle API, but no one has created a PR yet
  - @Lokathor will send a request to the glfw and winit maintainers to incorporate the raw-window-handle API
  - @navi mentioned that there is still an open issue with mac support (https://github.com/rust-windowing/raw-window-handle/issues/14)

* start a discussion about how to improve the WGs process, e.g. issue tracking, what information should be on the WG github, having only two maintainers, etc. (@Wodann ). @kvark proposes to have the discussion in a github issue

**Comment by msiglreith** [comment]
> navi (Github handle?)

hey, that's me 👋

**Comment by Lokathor** [comment]
update, GLFW _already_ supports raw-window-handle it seems, https://github.com/PistonDevelopers/glfw-rs/pull/421

**Comment by Lokathor** [comment]
Next issue meeting is here, and we have the minutes in, so I'm closing this https://github.com/rust-gamedev/wg/issues/59

I'll try to upload the recording within the next day.

**Comment by icefoxen** [comment]
Can't find the "https://github.com/amethyst/amethyst/wiki/Other-Game-Engines-in-Rust" page anywhere in this repo; if I hit the `wiki` link for this repo it just brings me back to the main page.  Permission issue?  It needs to be brought up to date anyway; I can take care of that if people want.

I REALLY WANT to volunteer for cpal performance improvements but it is among one of many things I can't commit to just now...

**Comment by Wodann** [comment]
@icefoxen why not just create PRs? Or is there no longer a maintainer for cpal?

**Comment by Lokathor** [comment]
"among one of many things I can't commit to just now" implies, to me, lack of time more than anything else.

**Comment by Wodann** [comment]
Ah, ambiguity..

**Comment by icefoxen** [comment]
Yep, I meant it as @Lokathor said.
